modules/db/src/main/Env.scala

```diff
@@ -1,37 +1,50 @@
 package lila.db
 
 import com.typesafe.config.Config
-import reactivemongo.api._
+import reactivemongo.api.{ DefaultDB, MongoConnection, MongoDriver }
 import scala.concurrent.duration._
-import scala.concurrent.Future
-import scala.util.{ Success, Failure }
-import dsl._
+import scala.concurrent.{ Await, ExecutionContext, Future }
+import scala.util.{ Failure, Success, Try }
+import dsl.Coll
 
 final class Env(
     config: Config,
     lifecycle: play.api.inject.ApplicationLifecycle) {
 
-  lazy val db = {
-    val parsedUri: MongoConnection.ParsedURI =
-      MongoConnection.parseURI(config.getString("uri")) match {
-        case Success(parsedURI) => parsedURI
-        case Failure(e)         => sys error s"Invalid mongodb.uri"
-      }
+  lazy val (connection, dbName) = {
     val driver = new MongoDriver(Some(config))
-    val connection = driver.connection(parsedUri)
 
-    parsedUri.db.fold[DefaultDB](sys error s"cannot resolve database from URI: $parsedUri") { dbUri =>
-      val db = DB(dbUri, connection)
-      registerDriverShutdownHook(driver)
-      logger.info(s"""ReactiveMongoApi successfully started with DB '$dbUri'! Servers: ${parsedUri.hosts.map { s => s"[${s._1}:${s._2}]" }.mkString("\n\t\t")}""")
-      db
-    }
+    (new Throwable).printStackTrace()
```

Comment:
Add to get a stack trace to check whether there is no hidden lifecycle issue (creating too much `Env` instances), or otherwise to make sure that something outside the sources is creating a `MongoDriver` (probably not in the appropriate way)